cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/ReplicationStatusKey.java

```diff
@@ -26,18 +26,28 @@
  */
 public final class ReplicationStatusKey {
 
+  private final String rowType;
+  private final String regionName;
+  private final String rsID;
   private final byte[] key;
 
   public ReplicationStatusKey(byte[] rowType, byte[] regionName, byte[] regionServerID) {
```

Comment:
this should take String, String, UUID so that the user can't pass in invalid types.

For example, it is currently buggy. You are passing in the byte representation of a UUID, but decoding those bytes as a string. This wouldn't happen if the constructor takes the actual type.